### DISCUSSION TITLE
Will this framework work with Google Cloud Functions / AWS Lambda / Azure Functions?

### DISCUSSION THREAD

**Discussion by piavgh** [fastapi/fastapi#7868] [Questions] [answered]
### First check

* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.

### Description

How can I use this framework with some serverless solutions out there (Google Cloud Functions / AWS Lambda / Azure Functions, ...)?

Thank you.

### Additional context

**Comment by koxudaxi** [comment, marked as answer]
@piavgh 
FastAPI supports ASGI protocol to connect a server.

The libraries are ASGI adapter which helps to run FastAPI on the services.

For AWS Lambda: 
https://github.com/erm/mangum
For Azure Functions: 
https://github.com/erm/bonnette
For Google Cloud Functions:
https://github.com/erm/grevillea

Example for AWS Lambda with mangum:
https://github.com/erm/mangum-examples

**Reply by RazCrimson** [reply]
Just an update, the above mentioned lib for GCP is currently not maintained.

GCP doesn't have support. But there are workarounds (check the below issue).

Issue tracked here: https://github.com/GoogleCloudPlatform/functions-framework-python/issues/194

**Comment by piavgh** [comment]
@koxudaxi : thank you. I will check those repos

**Comment by paul121** [comment]
Zeit now supports an [ASGI Interface for Python](https://zeit.co/docs/runtimes#advanced-usage/advanced-python-usage/asynchronous-server-gateway-interface).

It was fairly easy to set this up & deploy. Doesn't require making modifications to the application structure. Not sure if this would work well for most deployments (I think WebSockets would be out of question) but may be a good solution for simple apps!

I created a demo repo showcasing what I've found. Hope it can help anyone else that looks into FastAPI + Serverless: https://github.com/paul121/fastapi-zeit-now

**Comment by tiangolo** [comment]
Thanks for the help here @koxudaxi and @paul121 ! :cake: :bowing_man: 

Thanks @piavgh for reporting back and closing the issue :+1:

**Comment by dhimmel** [comment]
# GCP Cloud Functions

Noting the `vellox` python package for using fastapi with Google Cloud functions: [github](https://github.com/junah201/vellox), [docs](https://github.com/junah201/vellox), [pypi](https://pypi.org/project/vellox/)
